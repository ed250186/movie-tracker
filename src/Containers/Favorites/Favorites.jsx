import React, { Component } from "react";
import MovieCard from "../MovieCard/MovieCard.jsx";
import { connect } from "react-redux";
import { fetchFavoriteMovies } from "../../apiCalls/apiCalls";
import { addFavoriteMovie } from "../../actions/favoriteAction";

export class Favorites extends Component {
  constructor(props) {
    super(props);
    this.state = {
      displayFavMovies: false,
      error: ""
    };
  }
  displayFavoriteMovies = movies => {
    return movies.map(movie => {
      return <MovieCard 
        {...movie} 
        key={movie.id} 
        title={movie.title} 
        />
    })
  };

  findFaves = () => {
    const { favorites, movies } = this.props;
    console.log('fav', favorites)
    console.log(this.props)
    const faves = favorites.map(fav => {
      return movies.find(movie => {
        return movie.title === fav.title
      })
    })
    return this.displayFavoriteMovies(faves);
  }


  render() {
    
    return (
    <article>
      {!this.props.displayFavMovies && this.displayFavoriteMovies(this.props.movies)}
      {this.props.displayFavMovies && this.findFaves()}
    </article>
    );
  }
}

const mapStateToProps = state => ({
  favorites: state.favorites,
  movies: state.movies
});


export default connect(
  mapStateToProps
)(Favorites);
