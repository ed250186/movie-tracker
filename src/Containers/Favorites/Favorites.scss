
.movieCard {
  height: 250px;
  width: 150px;
  margin: 30px;
  position: relative;
  .card-img{
    height: auto;
    width: 150px;
    box-shadow: 2px 2px 10px black;
  }
  .inactive{
    position: absolute;
    z-index: 1;
    height: auto;
    width: 30px;
    margin-top: 5px;
    left: 105px;
  }
  .title{
    color: #F4F8FA;
    margin: 5px 0 5px 0;
  }
  .year{
    color: #49597B;
  }
}

.infoCard {
  width: 400%;
  position: absolute;
  z-index: 5;
  background: #314467;
  top: -5%;
  left: -5%;
  box-shadow: 2px 2px 10px black;
  padding: 10%;
  img {
    width: 100%;
    height: auto;
    margin-bottom: 5%;
  }
  .title {
    font-size: 1.3rem;
  }
  .overview{
    color: white;
    font-size: .9rem;
  }
}
