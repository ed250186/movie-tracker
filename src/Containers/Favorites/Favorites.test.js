import React from 'react';
import {shallow} from 'enzyme';
import {Favorites} from './Favorites';


describe('Favorites', () => {
  let wrapper, props
  beforeEach(() => {
    props = {
      favorites:[],
      movies: []
    }
    wrapper = shallow(<Favorites {...props}/>)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })


})