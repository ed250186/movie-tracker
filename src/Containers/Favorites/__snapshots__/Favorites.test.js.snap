// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot 1`] = `<article />`;
